### PR TITLE
Added cloud-config example into VMWare, cleaned up links in bare-metal doc

### PR DESCRIPTION
**Pull request by kayrus**
/cc @robszumski @joshix

### CODE REVIEW

File: os/booting-on-vmware.md [hunk 1]
```diff
@@ -76,7 +76,38 @@ Cloud-config data can be passed into a VM by attaching a [config-drive][config-d
...
+    # listen on both the official ports and the legacy ports
+    # legacy ports can be omitted if your application doesn't depend on them
+    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
+    listen-peer-urls: http://$private_ipv4:2380
+  units:
+    - name: etcd2.service
+      command: start
+    - name: fleet.service
+      command: start
+    - name: flanneld.service
```
Comment (robszumski) [inline os/booting-on-vmware.md]: for consistency with the rest of our cloudconfig examples, i'd say don't set up flannel

**Comment by robszumski** [comment]
LGTM otherwise. Thanks for cleaning this up.

**Comment by kayrus** [comment]
@robszumski done

**Comment by robszumski** [comment]
Thanks!

**Pull request merged**